src/plugins/apim/azureApimPlugin.ts

```diff
@@ -2,12 +2,26 @@ import Serverless from "serverless";
 import { ApimService } from "../../services/apimService";
 import { AzureBasePlugin } from "../azureBasePlugin";
 
-export class AzureApimServicePlugin extends AzureBasePlugin {
+export class AzureApimPlugin extends AzureBasePlugin {
 
   public constructor(serverless: Serverless, options: Serverless.Options) {
     super(serverless, options);
     this.hooks = {
-      "after:deploy:deploy": this.deploy.bind(this)
+      "after:deploy:deploy": this.deploy.bind(this),
+      "deploy:apim:apim": this.deploy.bind(this),
+    };
+
+    this.commands = {
+      deploy: {
+        usage: "Deploys APIM",
+        lifecycleEvents: ["deploy"],
```

Comment:
I think this is what's causing the deployment to run twice.

Reply:
If we want `sls deploy apim` to be an available command, I think we'd need to add it as a sub-command to the `AzureDeployPlugin` rather than the APIM plugin

Reply:
I'll remove this and try again - i mostly put it in to aid testing